### PR TITLE
 Fix logger port overwritten by batcher issue and add logger e2e test

### LINKED ISSUE #564: Basic Logger: Prediction didn't trigger message dumper
Fllowing doc[ here](https://github.com/kubeflow/kfserving/blob/master/docs/samples/logger/basic/README.md) 

Prediction didn't trigger message dumper, as below:
```
curl -v -H "Host: ${SERVICE_HOSTNAME}" http://$CLUSTER_IP/v1/models/$MODEL_NAME:predict -d $INPUT_PATH -H "Content-Type: application/json"
*   Trying 9.21.53.162...
* TCP_NODELAY set
* Connected to 9.21.53.162 (9.21.53.162) port 80 (#0)
> POST /v1/models/sklearn-iris:predict HTTP/1.1
> Host: sklearn-iris.default.mydomain.com
> User-Agent: curl/7.54.0
> Accept: */*
> Content-Type: application/json
> Content-Length: 76
>
* upload completely sent off: 76 out of 76 bytes
< HTTP/1.1 200 OK
< content-length: 23
< content-type: text/html; charset=UTF-8
< date: Wed, 20 Nov 2019 08:47:18 GMT
< server: istio-envoy
< x-envoy-upstream-service-time: 81
<
* Connection #0 to host 9.21.53.162 left intact
{"predictions": [1, 1]}%
```
Get the message dumper pod, it's not triggerred:
```
$ kubectl get pod -l serving.knative.dev/service=message-dumper
No resources found.

$ kc get deploy
NAME                                              READY   UP-TO-DATE   AVAILABLE   AGE
message-dumper-zxdjw-deployment                   0/0     0            0           8m43s .   
sklearn-iris-predictor-default-tktt2-deployment   1/1     1            1           2m55s
```

Notes: I can trigger message dumper successfully by manually, which mean, message dumper in knative works well.

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
The recent batcher change broke the request/response logger since it overwrites the container spec and the logger port change got lost.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #564 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by pugangxa** [comment]
Code seems good to me, but the test failed for test_logger.py. May need check whether we really get the "msg_dumper" pod in list_namespaced_pod.

File: pkg/logger/worker.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func (W *Worker) sendCloudEvent(logReq LogRequest) error {
 
 	t, err := cloudevents.NewHTTPTransport(
 		cloudevents.WithTarget(logReq.Url.String()),
-		cloudevents.WithEncoding(cloudevents.HTTPBinaryV1),
+		cloudevents.WithEncoding(cloudevents.HTTPBinaryV01),
```
Comment (cliveseldon) [inline pkg/logger/worker.go:58]: Why the change back to 01 when V1 is the released version?

Comment (yuzisun) [reply pkg/logger/worker.go:58]: we are using cloudevent sdk-go 1.2.0 which does not have HTTPBinaryV1 and caused test failure, which version of sdk-go we should use?

Comment (yuzisun) [reply pkg/logger/worker.go:58]: hmm actually I have some old vendor lingering around, there is `HTTPBinaryV1` but I still get this error
```
{"level":"error","ts":1594634991.9612372,"logger":"entrypoint","msg":"Failed to send log","URL":"http://message-dumper-predictor-default.kfserving-ci-e2e-test","error":"while sending event: error sending cloudevent: 400 Bad Request","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tgithub.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/kubeflow/kfserving/pkg/logger.(*Worker).Start.func1\n\tgithub.com/kubeflow/kfserving/pkg/logger/worker.go:109"}
```

Comment (yuzisun) [reply pkg/logger/worker.go:58]: it works with `HTTPBinaryV01` though, I am now confused.

Comment (yuzisun) [reply pkg/logger/worker.go:58]: ok confirmed that the issue is caused by the stale event display image which should be `gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display` and we still should us `HTTPBinaryV1`

**Pull request merged**